### ISSUE TITLE
CA1823 incorrectly reported for constant used in fixed-size buffer declaration

### ISSUE THREAD

**Issue by f2bo** [dotnet/roslyn-analyzers#6026]
### Analyzer

**Diagnostic ID**: [CA1823](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1823): `Avoid unused private fields`

### Analyzer source

**Version**: [SDK 6.0.300](https://dotnet.microsoft.com/download/dotnet/6.0)

(also reproduced with NuGet: `Microsoft.CodeAnalysis.NetAnalyzers: 7.0.0-preview1.22217.1`)

### Describe the bug

The following code incorrectly reports that the `MaxDataLength` constant is unused. 

### Steps To Reproduce

Enable `<AnalysisModePerformance>All</AnalysisModePerformance>` in the project file.

```C#
public unsafe struct Foo
{
    private const int MaxDataLength = 1024; // <== CA1823  Unused field 'MaxDataLength'

    private fixed byte _data[MaxDataLength];

    public byte[] GetData()
        => System.Runtime.InteropServices.MemoryMarshal.CreateSpan(ref _data[0], data_size).ToArray();

    public int data_size { get; set; }
}
```

### Expected behavior
No warning.

### Actual behavior
`CA1823  Unused field 'MaxDataLength'`   reported.

**Comment by Youssef1313** [comment]
Duplicate of https://github.com/dotnet/roslyn/issues/37593
